src/renderer/components/outputs/TextOutput.tsx

```diff
@@ -0,0 +1,58 @@
+import { Center, Flex, Spacer, Text } from '@chakra-ui/react';
+import { memo, useEffect } from 'react';
+import { useContext, useContextSelector } from 'use-context-selector';
+import { StringLiteralType } from '../../../common/types/types';
+import { isStartingNode } from '../../../common/util';
+import { GlobalContext, GlobalVolatileContext } from '../../contexts/GlobalNodeState';
+import { TypeTag } from '../TypeTag';
+import { OutputProps } from './props';
+
+export const TextOutput = memo(({ label, id, outputId, schemaId, useOutputData }: OutputProps) => {
```

Comment:
What's the reason for this component? The only difference between TextOutput, DirectoryOutput, and GenericOutput is how they handle broadcast data. They are identical otherwise.

Reply:
Exactly

Reply:
"Exactly" what lol 

You just duplicated a bunch of code everywhere. Why didn't you just give GenericOutput a prop like `convertOutputToType?: (value: unknown) => Type | undefined`. Or to be more compatible with the whole pick component thingy, you could also have made a higher kind component that is generic over the prop I suggested. So e.g.
```ts
const OutputComponent = (convertOutputToType?: (value: unknown) => Type | undefined) => {
  return memo(({ ... }: OutputProps) => { 
    // code of GenericOutput, but handling broadcasts like TextOutput with convertOutputToType 
  })
}

const TextOutput = OutputComponent(toStringType);
const DirectoryOutput = OutputComponent(toDirectoryType);
const GenericOutput = OutputComponent();
```

Reply:
I like that idea. I'll do that.

Reply:
Ended up not doing it that way because I felt like it was too complicated and we already pass the kind into the full props, so might as well just handle it in the type setter function itself